### PR TITLE
Feature/fwf 3756 edit js optimization

### PR DESCRIPTION
**Pull request by shuhaib-aot**
# Issue Tracking

JIRA: https://aottech.atlassian.net/browse/FWF-3756
Issue Type: BUG/ FEATURE

# Changes

# Screenshots (if applicable)

# Notes

# Checklist
- [ ] Updated changelog
- [ ] Added meaningful title for pull request

### CODE REVIEW

**Review by Bonymol-aot — APPROVED** [review]

File: forms-flow-web/src/components/CustomComponents/MultiSelect.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  return (
+    <div className="w-100">
+      <div className="input-with-pills form-control cursor-pointer position-relative">
+        {selectedRoles.map((role, index) => (
+          <CustomPill
+            key={role + index}
+            label={role}
+            icon={true}
+            bg="primary"
+            onClick={() => removeRole(role)}
```
Comment (fahad-aot) [inline forms-flow-web/src/components/CustomComponents/MultiSelect.js:67]: please add aria-label and data-testid.

File: forms-flow-web/src/components/CustomComponents/MultiSelect.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          />
+        ))}
+        <input
+          type="text"
+          value={roleInput}
+          onChange={handleRoleInputChange}
+          onFocus={() => setIsDropdownOpen(true)} // Open dropdown when input is focused
+          className="role-input"
+          placeholder="Type to filter roles"
+          ref={inputRef} // Reference to input for outside click handling
```
Comment (fahad-aot) [inline forms-flow-web/src/components/CustomComponents/MultiSelect.js:77]: Also here.

File: forms-flow-web/src/components/CustomComponents/MultiSelect.js (hunk 1 above)
Comment (arun-s-aot) [reply forms-flow-web/src/components/CustomComponents/MultiSelect.js:67]: This we can avoid for now as we are planning to completely change this MultiSelect component @fahad-aot

File: forms-flow-web/src/components/CustomComponents/MultiSelect.js (hunk 2 above)
Comment (arun-s-aot) [reply forms-flow-web/src/components/CustomComponents/MultiSelect.js:77]: This we can avoid for now as we are planning to completely change this MultiSelect component @fahad-aot

**Review by arun-s-aot — APPROVED** [review]

**Pull request merged**